backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/handler/ExceptionHandler.kt

```diff
@@ -0,0 +1,113 @@
+package com.nosql.personservice.common.exception.handler
+
+import com.nosql.personservice.common.exception.dto.ApplicationErrorDto
+import com.nosql.personservice.common.exception.enumerator.ApplicationExceptionCodeEnum.DECODING_ERROR
+import com.nosql.personservice.common.exception.enumerator.ApplicationExceptionCodeEnum.VALIDATION_ERROR
+import com.nosql.personservice.common.exception.ApplicationException
+import com.nosql.personservice.common.logger.logger
+import org.slf4j.Logger
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.support.WebExchangeBindException
+import org.springframework.web.server.ServerWebInputException
+import java.util.StringJoiner
+import java.util.stream.Collectors
+
+@ControllerAdvice
+class ExceptionHandler {
+
+    private val log: Logger by logger()
+
+    @ExceptionHandler(ApplicationException::class)
+    fun handleCommonException(exception: ApplicationException): ResponseEntity<ApplicationErrorDto> {
+
+        val errorDto = ApplicationErrorDto(exception)
+        val status = exception.status
+
+        log.error("Got application error. Response: $errorDto")
+
+        return ResponseEntity
+            .status(status ?: HttpStatus.INTERNAL_SERVER_ERROR)
+            .body(errorDto)
+    }
+
+    @ExceptionHandler(WebExchangeBindException::class)
+    fun handleException(exception: WebExchangeBindException): ResponseEntity<ApplicationErrorDto>? {
+
+        val errorMessageJoiner = StringJoiner(ERROR_MESSAGE_DELIMITER)
+        val bindingResult = exception.bindingResult
+
+        if (bindingResult.hasGlobalErrors()) {
+
+            val globalErrorsMessages = bindingResult.globalErrors.stream()
+                .map { String.format("%s: %s", it.objectName, it.defaultMessage) }
```

Comment:
kotlin string template
copy this class from auth-service

Reply:
fixed